### ISSUE TITLE
utcar running in docker on NAS, trying to catch the heartbeat... 

### ISSUE THREAD

**Issue by jnathalia** [tdeckers/utcar#6]
I'm trying things in a few steps:
1. Running utcar in a docker and getting the heartbeat (and more if that works)
2. Redirect the results of step 1 to my Home Assistant with SIA integration

Step 1.
I have installed the docker utcar on my Synology NAS. It's working and listening on port 12300
My NAS is on IP 192.168.1.100. The docker isn't bridged but using the host IP address 192.168.1.100.

![Screenshot 2024-01-18 at 15 40 30](https://github.com/tdeckers/utcar/assets/3389896/4ecadc4d-dae8-4b58-ab0e-ac9ac07079ed)

My ATS-IP-KIT (ATS1809) is at IP 192.168.1.202. If I enter this address in my browser, I can make the settings to which Central Station how it should communicate. See below:

![Screenshot 2024-01-18 at 15 33 42](https://github.com/tdeckers/utcar/assets/3389896/155c52ee-76e1-43dc-939a-131081a730c0)

At my ATS alarm system, I have defined a PAC (3, NAS utcar) to communicate for the Area's by defining which account number should be communicated (in my case 1001).

![Screenshot 2024-01-18 at 15 48 26](https://github.com/tdeckers/utcar/assets/3389896/1a9a7f78-c069-4d58-958c-65a8e1373f6c)

The problem is, I don't receive a heartbeat which I have set to every 10 seconds. 

Supposing the log (3rd tab in the container) should show the heartbeat, nothing happens...
What am I doing wrong?

If anybody can get me on the right track I would be thankful.

**Comment by bjornmorsman** [comment]
Hi tnathalia,

Does your system report to pac 3? You can enable or disable the reports to pac 1, 2, 3, 4

**Comment by jnathalia** [comment]
Yes, I do. Please see next (sorry the Titan software is in Dutch, but it basically says that this particular Area will communicate to PAC 2 and PAC 3):

![Screenshot 2024-01-19 at 17 18 10](https://github.com/tdeckers/utcar/assets/3389896/3588b88d-01b5-484a-ab18-ad628f2e1836)

All other defined Areas (basically used for my main alarm system, all the 4 PAC have been ticked on. So, it any communication which goes to my external PAC should go anyhow to PAC 2 (Home Assistant) en PAC 3 (docker NAS utcar) as well.

But besides from this, at my ATS-IP-KIT network card itself you can set the Heartbeat. So, the least it should provide me this I would guess...

**Comment by bjornmorsman** [comment]
No problem for the dutch i am dutch myself. 

So that is not the problem, so in you docker log you dont get any error of some kind? If i send data from my advanced to utcar and utcar doesnt understand the message i get a log entry with “wrong sia code”

I think the problem is in the handeling of the software of utcar, ats master (the system your have) is very different of the ats advanced, i think the code is writen for the ats advanced.

maybe you can use the printer port of your ats1801 and convert that info with an serial to mqtt converter? The only downside of that is that you need to decrypt en translate the messages from the ats, so that you know what is going on at the ats side.

**Comment by jnathalia** [comment]
this is all I get... (or I'm looking at the wrong place?):

![Screenshot 2024-01-18 at 15 40 30](https://github.com/tdeckers/utcar/assets/3389896/bbbecd55-2a23-4755-9fd1-97e9a40308ec)

I have an ATS4000 Classic, the newer one is the ATS4500 (Advanced) but both support SIA of course.

But if it is the software of utcar (or even SIA in Home Assistant), is there a possibility to catch the text that is sent to port 12300? For instance by a network scanner or so?

Going down the route to ATS1801 is even more difficult as the ATS supports this only:

![Screenshot 2024-01-19 at 18 16 20](https://github.com/tdeckers/utcar/assets/3389896/88e8c53f-2d87-45b9-adde-1dd9c3fd5338)

And I think I'm at the wrong place, as this is only for the Titan software how to connect to the ATS...

**Comment by jnathalia** [comment]
I have been thinking the whole night about it, but I'm triggered about your comment @bjornmorsman "i think the code is writen for the ats advanced"...

Is the code of utcar module specifically listening to an ATS Advanced and/or Ajax alarm system based on SIA? Probably the ATS Classic gives just a small modification or different format of the SIA code and it's not being parsed by the utcar module? Same happens for the Home Assistant SIA add-on probably.

I think the only person who knows is @tdeckers I believe.... Are you possibly available?

Alternatively, I will test with Curl to PUT some text to the IP over the port 12300 and see if that comes through...

**Comment by jnathalia** [comment]
![Screenshot 2024-01-20 at 14 57 49](https://github.com/tdeckers/utcar/assets/3389896/b8e38eca-6e7e-4922-bf78-e1e647168b28)

I received something this morning... I was playing around with some settings, but later this afternoon I didn't know what I did. It shows something, that it cannot interpret the message. So, @bjornmorsman it looks like you are right...

**Comment by jnathalia** [comment]
I found out that the log was created due to a curl -X POST command I have been testing with. So, it had nothing to do with an ATS setting or message that came through...

**Comment by dIP365** [comment]
Hi!
I have played little with my ATS3007, and got nothing in my utcar log, until i sync'ed the ATS's time with my computer
looks like the alarm wa several minutes away :)
now i get following when i try to disable a zone, so i need to change the .go code to understand my ATS

-----------------------------------------------------------
utcar    | 2024/02/11 18:37:30 Message:  01010036"SIA-DCS"0050R0001L0001[#005555|NOP000*'User 00000'NM]
utcar    | 2024/02/11 18:37:30 {{13937060687746241641 22774721411493 0xc90400} 0050 0001 0001 005555 OP 000}
utcar    | 2024/02/11 18:37:30 Push error: Unsupported SIA command for pusher (OP)

**Comment by dIP365** [comment]
i found this : 
utcar | 2024/02/11 18:37:30 {{13937060687746241641 22774721411493 0xc90400} 0050 0001 0001 005555 OP 000}
005555
 ^^^^------------is zone 
and "OP 000" is like "open" or when you deactivate a zone and the 000 is usernumber 000=admin (from titan)
CL is when you activate a zone

**Comment by martinkura-svk** [comment]
If it helps anyone, in ATS1809 you need to set the encryption to Triple DES, then it will start working with UTCAR.
